docs/datatypes/io.md

```diff
@@ -161,3 +161,29 @@ val action: Task[String] = Task.effectTotal(i += 1) *> Task.fail(new Throwable("
 val cleanupAction: UIO[Unit] = UIO.effectTotal(i -= 1)
 val composite = action.ensuring(cleanupAction)
 ```
+### A full working example on using brackets
+```scala mdoc:silent
+
+import scalaz.zio.{App, Task, UIO}
+import java.io.{File, FileInputStream}
+import java.nio.charset.StandardCharsets
+
+object Main extends App {
+
+  // run my bracket
+  def run(args: List[String]) =
+    mybracket.orDie.const(0) 
+
+  def closeStream(is: FileInputStream) =
+    UIO(is.close())
+
+  def convertBytes(is: FileInputStream) =
+    Task.effect(println(new String(is.readAllBytes(), StandardCharsets.UTF_8))) // Java 11+, since is.realAllBytes() won't compile on Java8
```

Comment:
Since this won't work on Java 8, I'd suggest writing a helper, a private method inside `convertBytes`:

```scala
def readAll(fis: FileInputStream, len: Long): Array[Byte] = {
  val content = Array.ofDim(len)
  fis.read(content)
  content
}
```

You can get the length of the file using `new File(path).size` or something similar (can't remember it off the top of my head).